src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -811,6 +811,10 @@ internal static void ValidateFunction(FunctionDescriptor function, Dictionary<st
 
                 httpFunctions.Add(function.Name, httpTrigger);
             }
+            if (environment != null && environment.IsFlexConsumptionSku() && function.Metadata.IsLegacyBlobTriggerFunction())
+            {
+                throw new InvalidOperationException($"Regular blob trigger for function '{function.Name}' is not supported in Flex Consumption plan. Please change the blob trigger to use Event Grid.");
```

Comment:
Suggest an error message like "The Flex Consumption SKU only supports EventGrid as the source for BlobTrigger functions. Please update function X to use EventGrid."

Also, to Shyju's comment - perhaps we should also have an aka.ms link to point to the relevant doc page https://learn.microsoft.com/en-us/azure/azure-functions/functions-event-grid-blob-trigger?pivots=programming-language-csharp

Reply:
Corrected the text and added the link.